test/opus/pipeline/stage_test.exs

```diff
@@ -283,6 +283,37 @@ defmodule Opus.Pipeline.StageTest do
     end
   end
 
+  describe ":error_message option, with a function" do
+    defmodule ErrorMessages do
+      def invalid(input), do: "Invalid input: #{input}"
+    end
+
+    defmodule ErrorMessageWithFunctionPipeline do
+      use Opus.Pipeline
+
+      step :double
+      step :maybe_fail,
+        error_message: fn n ->
+          "Can't work with #{n}"
+        end
+      step :fail_for_sure, error_message: &ErrorMessages.invalid/1
```

Comment:
I'd prefer defining the `fail_for_sure` function and raise with `raise/1`. Otherwise the reader has think that this does indeed fail for sure with `UndefinedFunctionError` because the function is undefined.